metaseq/hub_utils.py

```diff
@@ -678,6 +678,7 @@ def generate(
                     beams.append(result)
                 retval.append(beams)
 
+        torch.cuda.synchronize()
```

Comment:
Nit: add a comment for why this sync was added / what happens to generated output if it gets removed?